storage-api/src/main/java/com/connexta/transformation/storage/api/status/MetadataInfo.java

```diff
@@ -0,0 +1,102 @@
+ */
+public interface MetadataInfo extends TransformationStatus {
+
+  /**
+   * Returns the type of metadata being created (e.g. metacard, irm...).
+   *
+   * @return the type of metadata being created
+   */
+  String getMetadataType();
+
+  /**
+   * Returns an optional {@link InputStream} for retrieving the metadata. If the metadata hasn't
+   * been created yet, or failed to be created, the optional will be empty.
+   *
+   * @return an optional of the input stream for this metadata
+   */
+  Optional<InputStream> getContent();
+
+  /**
+   * Returns an optional containing the content type of the metadata. If the metadata hasn't been
+   * created yet, or failed to be created, the optional will be empty.
+   *
+   * @return an optional containing the content type of the metadata
+   */
+  Optional<String> getContentType();
+
+  /**
+   * Returns an optional containing the length of the metadata. If the metadata hasn't been created
+   * yet, or failed to be created, the optional will be empty.
+   *
+   * @return an optional containing the length of the metadata
+   */
+  Optional<String> getContentLength();
+
+  /**
+   * Used to signify the successful creation of the metadata this MetadataInfo represents, this
+   * method will persist the given content and mark this metadata transformation as complete. The
+   * given {@link InputStream} will be read and closed within this method.
+   *
+   * @param contentType the content type of the metadata
+   * @param contentStream an {@link InputStream} for reading the metadata
+   */
+  void complete(String contentType, InputStream contentStream);
+
+  /**
+   * Used to signify that the creation of this metadata failed, this method will mark this metadata
+   * as failed and will persist a marker to indicate that.
+   *
+   * @param reason an {@link ErrorCode} indicating why the metadata creation failed.
+   */
+  void fail(ErrorCode reason);
+
+  /**
+   * Used to signify that the creation of this metadata failed, this method will mark this metadata
+   * as failed and will persist a marker to indicate that.
+   *
+   * @param reason an {@link ErrorCode} indicating why the metadata creation failed.
+   * @param message a message including information about why the metadata creation failed.
+   */
+  void fail(ErrorCode reason, String message);
+
+  /**
+   * Returns an optional containing an {@link ErrorCode} indicating why this metadata failed to be
+   * created. If the creation is still in progress or the metadata was created successfully, the
+   * optional will be empty.
+   *
+   * @return an optional containing an {@link ErrorCode} indicating why this metadata failed to be
+   *     created.
+   */
+  Optional<ErrorCode> getFailureReason();
+
+  /**
+   * Returns an optional containing a message indicating why this metadata failed to be created. If
+   * the creation is still in progress or the metadata was created successfully, the optional will
+   * be empty.
```

Comment:
✏️ If we keep the method for reporting a failure without a message (I don't think we should), then a third scenario where the optional will be empty is if the consumer didn't give a message.